### ISSUE TITLE
autoupload truncates files

### ISSUE THREAD

**Issue by Ventusfahrer** [nextcloud/android#399]
### Actual behaviour
I have defined an autoupload on screenshots of my Nexus 9.

If I take a screenshot, the file is going to be uploaded.
But the uploaded file is smaller than the original one (48 vs 1791015 Bytes).
When using the file upload functionality (Upload in "all files") the complete file is uploaded.

### Expected behaviour
files must be uploaded complete
 
### Steps to reproduce
1. define autoupload on screenshots
2. create a screenshot
3. check the uploaded file 

### Environment data
Android version:
7.0 Security Update from  5th November 2016

Device model: 
Nexus 9

Stock or customized system:
?
Nextcloud app version:
1.4.0 RC2
Nextcloud server version:
10.0.1
#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

**Comment by Ventusfahrer** [comment]
Hi Andy, 

maybe the attachment is of interest for you.
I did play around with the AutoUpload feature and did see that obviously there are a lot of temporary files created by the Android Camera App (and probably by the screenshot too). 

![uploadprotocol](https://cloud.githubusercontent.com/assets/5901190/20709634/cb4f9610-b636-11e6-87c4-79f33e5937bd.png)

regards

Peter

**Comment by AndyScherzinger** [comment]
Hi Peter,

the empty IMG temp file might be related to https://code.google.com/p/android/issues/detail?id=220928 depending which version of Android 7+ you are running. For the videos though I have no explanation why that happens, maybe also some processing by Google - #397. The empty folder might also be related to HDR post-processing as described in #390 too.
@tobiasKaminsky's idea to wait for some seconds before actually trying to upload anything, see #403 might resolve this matter.

Best regards,
Andy

**Comment by Ventusfahrer** [comment]
let's hope that the post processing is finished within these _some seconds_. This may be the case for relative small photos, but for large videos?)

best regards

Peter

**Comment by AndyScherzinger** [comment]
Videos might be a problem yes :(
Haven't had a nice idea how to solve that part (not even on a conceptual level)

**Comment by Ventusfahrer** [comment]
what's about:
1. on event detection get the file attributes (last write time, size)
2. wait x seconds 
      is the file still there 
          has the file changed size or last write time since last check
                wait another x seconds
           else
                start upload
       else (no longer there --> assume the file is temporary file)
                skip the upload

hope this works, needs regular update on the file visible in the attributes.

**Comment by AndyScherzinger** [comment]
Might work yes, what i don't know is if and when Google is touching the files during post processing.

**Comment by tobiasKaminsky** [comment]
Good idea.
I will implement it in kind of this way and we can test it in beta if it solves the problem for image/video.
Doing this tomorrow afternoon.

**Comment by tobiasKaminsky** [comment]
Maybe there is also some kind of querying if the file is already opened for writing and wait until it closes...

**Comment by Ventusfahrer** [comment]
Hi, the Problem still exists in 1.4.0, -see screenshot

![screenshot_20161207-214703](https://cloud.githubusercontent.com/assets/5901190/20986065/51af85a8-bcc7-11e6-8fbc-bc00309ab3eb.png)

**Comment by AndyScherzinger** [comment]
Cc @tobiasKaminsky :(

**Comment by tobiasKaminsky** [comment]
I have implemented it currently with a "stupid" delay of 10s.
If you want, you can try the beta of tomorrow.

**Comment by AndyScherzinger** [comment]
the wait and the deleted file check have been implemented by @tobiasKaminsky and merged for 1.4.1, so I'll close the issue for now and it can then be tested via 1.4.1 RC1 as soon as it is released (still some open issues to be fixed).
